### ISSUE TITLE
Handle the max allowed refund for partially paid orders

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#18286]
Follow-up of this [topic](https://github.com/PrestaShop/PrestaShop/pull/18164#issuecomment-601341828)

@MatShir @colinegin we need your help on this
@jolelievre can explain further the issue if needed

**Comment by jolelievre** [comment]
More details about this issue:

Previously the refund feature was enabled only when the order had a status `paid` which is based only on order status not actual payments.

Now, the refund is possible when:
- order has status paid
or
- at least one payment exist on the Order

Which raises a new questions:

- when the order is partially paid and I want to refund an order what is the max authorized? and how is it dispatched on the different products?
- when I partially refund a product it is not refundable any more even if it has only been partially refunded, is it acceptable?

**Comment by sam-pires** [comment]
As discussed today with @jolelievre, 

TLDR: we keep things as they are.

>when the order is partially paid and I want to refund an order what is the max authorized? and how is it dispatched on the different products?

In legacy, there is no limit on the refund amount on a product on a partial refund. 
The maximum is the product price.

-> In order to not disturb/impact current processes used by merchants, we decided it is better to **not change this behavior.** 
We don't want to limit the merchants, they will be free to choose how much they want to refund, even if the refund is higher than the payments received at that time.
We keep the product price as a maximum.

<br>
<br>

>when I partially refund a product it is not refundable any more even if it has only been partially refunded, is it acceptable?

This behavior is the same in legacy.
We need more research and feedback to know if allowing multiple refunds on a product is worthy for merchants.
Here again, in order to not disturb/impact merchants' current processes we decided to keep the same behavior.


<br>
@MatShir do you validate?

**Comment by MatShir** [comment]
@jolelievre I agree with @sam-pires it seems the optimal choice here.

**Comment by colinegin** [comment]
We need this decision to be available in the specs. Thanks !

**Comment by jolelievre** [comment]
Ok then if we change nothing I guess we can update the specs close this issue ^^

**Comment by MatShir** [comment]
[Here](https://github.com/PrestaShop/prestashop-specs/commit/cd3974eb429f8a3df055621414ff790bc4cbf92f) the related specs.
Thanks you all :)
